### PR TITLE
Create a target shorthost length.

### PR DESCRIPTION
**Pull request by jof**
This is meant to address #7452 
```
jof@spacecuke ~/src/librenms % ./scripts/pre-commit.php
Running unit tests... success
Running style check... success
Running lint check... success
Tests ok, submit away :) 
```

- [ x ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

**Comment by murrant** [comment]
Would just removing  the mydomain setting from the hostname work?

**Comment by jof** [comment]
@murrant: Looking at the `shorthost()` function, it doesn't take into consideration anything to do with mydomain.
It splits up the `hostname` into DNS labels (split by ".") and concatenates the left-most labels until the maximum length is met or exceeded: https://github.com/librenms/librenms/blob/46ee1b377cf262022c311926b879b7649973cc98/includes/common.php#L83-L98

**Comment by murrant** [comment]
I am just trying to avoid adding a new setting that could cause unexpected results.

Many RRD layouts can be broken by having too long of string.

My suggestion was that removing the common part of the domain might be a better way to shorten.  Then to fix you particular issue, you would need to increase the default length.  Then go through the uses and set some back to 12.

File: includes/common.php [hunk 1]
```diff
@@ -86,6 +86,10 @@ function shorthost($hostname, $len = 12)
     if (filter_var($hostname, FILTER_VALIDATE_IP)) {
         return $hostname;
     }
+    global $config;
```
Comment (murrant) [inline includes/common.php]: This can just be:
```
   $len = Config::get('shorthost_target_length', $len);
```

File: includes/defaults.inc.php [hunk 2]
```diff
@@ -947,3 +947,6 @@
 $config['api']['cors']['maxage'] = '86400';
 $config['api']['cors']['allowmethods'] = array('POST', 'GET', 'PUT', 'DELETE', 'PATCH');
 $config['api']['cors']['allowheaders'] = array('Origin', 'X-Requested-With', 'Content-Type', 'Accept', 'X-Auth-Token');
+
+// The target maximum hostname length when applying the shorthost() function.
+$config['shorthost_target_length'] = 12;
```
Comment (murrant) [inline includes/defaults.inc.php]: Please comment this out.

Did you check if setting it higher messes up any pages/graphs?

Comment (jof) [reply includes/defaults.inc.php]: I am running this in my environment with strings up to 25 chars long with no problem.
I will comment this out and leave a bit more description

File: includes/defaults.inc.php [hunk 3]
```diff
@@ -947,3 +947,11 @@
 $config['api']['cors']['maxage'] = '86400';
 $config['api']['cors']['allowmethods'] = array('POST', 'GET', 'PUT', 'DELETE', 'PATCH');
 $config['api']['cors']['allowheaders'] = array('Origin', 'X-Requested-With', 'Content-Type', 'Accept', 'X-Auth-Token');
+
+// The target maximum hostname length when applying the shorthost() function.
```
Comment (laf) [inline includes/defaults.inc.php]: We need docs adding as well. If you can move all of these comments to https://github.com/librenms/librenms/blob/master/doc/Support/Configuration.md#webui-settings somewhere in that section that would be good.

After that, 👍 from me.

**Comment by laf** [comment]
@jof Any further update on this?

**Comment by jof** [comment]
Sorry for the delay, I've just been busy with other stuff.

That's a much better location for documentation. Thanks for the pointer!
I pushed an updated branch.

**Pull request merged**